### DISCUSSION TITLE
open telemetry integration

### DISCUSSION THREAD

**Discussion by brewsoftware** [OpenLineage/OpenLineage#1542] [Ideas]
Hey all, 
   So I raised this one some time ago but it would be great to connect this project with the open trace standard. https://github.com/open-telemetry/community/discussions/1319

Some way of manaul instrumentation from a "telemetry" point of view and maybe a connector for consuming open trace logs for various trace metrics. 

Given that open trace already has some notions of spans, is there a way of integrating the two standards. This could possibly work both ways instead of having two independent standards for slightly different use cases it would be cleaner to have a unified standard or upgrading both standards to incorporate the common elements.

Im somewhat new to open lineage so would be interested in feedback from the community.

**Comment by julienledem** [comment]
I think it makes a lot of sense to connect both standards. 
There are differences between data pipelines and services that cause the need for two models. But we should definitely connect both. We want the upstream lineage from data collection and traces. 
- Services typically directly call downstream services and the traces form a tree originating from a single upstream call.
- Data Pipelines pull data from upstream and produce data downstream. They typically don't invoke directly downstream consumers and data aggregation means this is a graph and not a tree like for spans. Lineage also typically doesn't go at the row level when traces are at the request level.

Some thoughts here: https://openlineage.io/blog/openlineage-takes-inspiration-from-opentelemetry/
I also spoke at a meetup about this (I start at the 30 min mark): https://www.youtube.com/watch?v=7kj-_Bd8TbE
